ImperatorToCK3/Source/CommonUtilities/History.cpp

```diff
@@ -0,0 +1,37 @@
+#include "ParserHelpers.h"
+#include "CommonRegexes.h"
+#include "History.h"
+
+
+SimpleField::SimpleField(const std::optional<std::string>& initialValue) : initialValue(initialValue) {}
+
+std::optional<std::string> SimpleField::getValue(const date& date) const {
+	if (const auto& lowerBound = valueHistory.lower_bound(date); lowerBound != valueHistory.end()) {
+		return lowerBound->second;
+	}
+	return initialValue;
+}
+
+
+
+DatedHistoryEntry::DatedHistoryEntry(std::istream& theStream)
+{
+	registerRegex(commonItems::stringRegex, [&](const std::string& key, std::istream& theStream) {
+		auto rightSide = commonItems::getString(theStream);
+		if (contents.contains(key)) [[unlikely]]
```

Comment:
```
auto [inserted, itr] =contents[key].emplace_back(rightSide);
if (!inserted)
  itr = rightSide;
```

This will ensure at most one lookup in the map. As written now, there will be two lookups.

Reply:
Dang, I forgot about the pattern you showed me a few days ago 😐

Reply:
This didn't work as contents[key] is a vector. I was able to simplify to just
```
registerRegex(commonItems::stringRegex, [&](const std::string& key, std::istream& theStream) {
		contents[key].emplace_back(commonItems::getString(theStream));
	});
```
. though.